src/ray/raylet/local_task_manager.cc

```diff
@@ -878,6 +883,7 @@ void LocalTaskManager::Dispatch(
   }
   worker->AssignTaskId(task_spec.TaskId());
   worker->SetAssignedTask(task);
+  worker->SetAncestorDetachedActorId(task_spec.AncestorDetachedActorId());
```

Comment:
Perhaps we can just do this in `SetAssignedTask`?

Reply:
Also is this only set while the worker is executing this task? I thought we would need to set it for the lifetime of the worker?

Reply:
Once set it won't be unset for it's for the lifetime of the worker.